caiman_napari/main_offline_gui.py

```diff
@@ -97,36 +128,72 @@ def add_item(self, algo: str, parameters: dict, name: str, input_movie_path: str
         item = self.ui.listWidgetItems.item(n - 1)
         item.setData(3, uuid)
 
+    def remove_item(self):
+        item_gui = QtWidgets.QListWidgetItem = self.ui.listWidgetItems.currentItem()
+        uuid = item_gui.data(3)
+
+        ix = self.dataframe[self.dataframe['uuid'] == uuid].index[0]
+        self.dataframe.caiman.remove_item(index=ix)
+        self.ui.listWidgetItems.takeItem(ix)
+
     def run_item(self):
         item_gui = QtWidgets.QListWidgetItem = self.ui.listWidgetItems.currentItem()
         uuid = item_gui.data(3)
 
         ix = self.dataframe[self.dataframe['uuid'] == uuid].index[0]
 
+        # For now, just run given index.
         self._run_index(ix)
 
     def run(self):
         self._run_index(0)
 
     def _run_index(self, index: int):
         callbacks = [partial(self.item_finished, index)]
+        std_out = self._print_qprocess_std_out
+
+        self.dataframe.iloc[index].caiman.run(callbacks_finished=callbacks, callback_std_out=std_out)
 
-        self.dataframe.iloc[index].caiman.run(callbacks_finished=callbacks)
+    def _print_qprocess_std_out(self, proc):
+        txt = proc.readAllStandardOutput().data().decode('utf8')
+        self.ui.textBrowserStdOut.append(txt)
 
     def item_finished(self, ix):
         self.set_list_widget_item_color(ix, 'green')
 
+        if (ix + 1) < self.ui.listWidgetItems.count():
+            time.sleep(10)
+            self._run_index(ix + 1)
+
+        else:
+            QtWidgets.QMessageBox.information(self, 'Batch is done!', 'Yay, your batch has finished processing!')
+
+    def set_params_text(self, ix):
+        p = self.dataframe.iloc[ix]['params']
+        print(p)
+        u = self.dataframe.iloc[ix]['uuid']
+        s = pprint.pformat(p)
+        s = f"{u}\n\n{s}"
+        self.ui.textBrowserParams.setText(s)
+
     def set_list_widget_item_color(self, ix: int, color: str):
         self.ui.listWidgetItems.item(ix).setBackground(QtGui.QBrush(QtGui.QColor(COLORS_HEX[color])))
 
     def show_cnmf_params_gui(self):
         self.cnmf_gui = CNMFWidget(parent=self)
-        # self.viewer.window.add_dock_widget(self.cnmf_gui)
         self.cnmf_gui.show()
 
     def show_mcorr_params_gui(self):
         self.mcorr_gui = MCORRWidget(parent=self)
         self.mcorr_gui.show()
+    def show_cnmf_results(self):
```

Comment:
This is a messy way to do it. Have one function like `load_output()` which then calls the appropriate module based on the `algo` for that dataframe item.

Reply:
before calling the corresponding `algo_module.load_output()` this function can call `self.open_movie()` to open the movie.